apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -45,9 +91,37 @@ const initialState = {
   // that are not in a course)
   primaryAssignmentIds: [],
   // Mapping from sectionId to section object
-  sections: {}
+  sections: {},
+  // We can edit exactly one section at a time.
+  // While editing we store that section's 'in-progress' state separate from
+  // its persisted state in the sections map.
+  sectionBeingEdited: null,
+  saveInProgress: false,
 };
 
+/**
+ * Generate shape for new section
+ * @param id
+ * @param courseId
+ * @param loginType
+ * @returns {sectionShape}
+ */
+function newSectionData(id, courseId, loginType) {
```

Comment:
This helper will go away when we ship new section flow and can tear out the old path.